libs/payments/eligibility/src/lib/eligibility.manager.ts

```diff
@@ -2,9 +2,88 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+import {
+  ContentfulManager,
+  EligibilityOfferingResult,
+} from '../../../../shared/contentful/src';
 import { Injectable } from '@nestjs/common';
 
+import { OfferingComparison, OfferingOverlapResult } from './eligibility.types';
+
 @Injectable()
 export class EligibilityManager {
-  constructor() {}
+  constructor(private contentfulManager: ContentfulManager) {}
+
+  /**
+   * Determine what existing planIds or Offerings a user has overlap with
+   * a desired target plan and what the comparison is to the target plan.
+   *
+   * @returns Array of overlapping planIds/offeringProductIds and their comparison
+   *          to the target plan.
+   */
+  async getOfferingOverlap(
+    planIds: string[],
+    offeringStripeProductIds: string[],
+    targetPlanId: string
+  ): Promise<OfferingOverlapResult[]> {
+    if (!planIds.length && !offeringStripeProductIds.length) return [];
+
+    const detailsResult =
+      await this.contentfulManager.getPurchaseDetailsForEligibility([
+        ...planIds,
+        targetPlanId,
+      ]);
+
+    const result: OfferingOverlapResult[] = [];
+
+    const targetOffering = detailsResult.offeringForPlanId(targetPlanId);
+    if (!targetOffering) return [];
+
+    for (const offeringProductId of offeringStripeProductIds) {
+      const comparison = offeringComparison(offeringProductId, targetOffering);
+      if (comparison) result.push({ comparison, offeringProductId });
+    }
+
+    for (const planId of planIds) {
+      const fromOffering = detailsResult.offeringForPlanId(planId);
+      if (!fromOffering) continue;
+      const comparison = offeringComparison(
+        fromOffering.stripeProductId,
+        targetOffering
+      );
+      if (comparison) result.push({ comparison, planId });
+    }
+    return result;
+  }
+}
+
+function offeringComparison(
+  fromOfferingProductId: string,
+  targetOffering: EligibilityOfferingResult
+) {
+  if (targetOffering.stripeProductId === fromOfferingProductId)
+    return OfferingComparison.SAME;
+  const commonSubgroups =
+    targetOffering.linkedFrom.subGroupCollection.items.filter(
+      (sg) =>
+        !!sg.offeringCollection.items.find(
+          (oc) => oc.stripeProductId === fromOfferingProductId
+        )
+    );
+  if (!commonSubgroups.length) return null;
```

Comment:
What does a null return value mean in the context of this function?

(an aside - would be nice to have a jsdoc on this fn)